lib/stytch/oauth.rb

```diff
@@ -1,54 +1,161 @@
+    #   The type of this field is nilable `String`.
+    # session_custom_claims::
+    #   Add a custom claims map to the Session being authenticated. Claims are only created if a Session is initialized by providing a value in `session_duration_minutes`. Claims will be included on the Session object and in the JWT. To update a key in an existing Session, supply a new value. To delete a key, supply a null value.
+    #
+    #   Custom claims made with reserved claims ("iss", "sub", "aud", "exp", "nbf", "iat", "jti") will be ignored. Total custom claims size cannot exceed four kilobytes.
+    #   The type of this field is nilable `object`.
+    # code_verifier::
+    #   A base64url encoded one time secret used to validate that the request starts and ends on the same device.
+    #   The type of this field is nilable `String`.
+    #
+    # == Returns:
+    # An object with the following fields:
+    # request_id::
+    #   Globally unique UUID that is returned with every API call. This value is important to log for debugging purposes; we may ask for this value to help identify a specific API call when helping you debug an issue.
+    #   The type of this field is `String`.
+    # user_id::
+    #   The unique ID of the affected User.
+    #   The type of this field is `String`.
+    # provider_subject::
+    #   The unique identifier for the User within a given OAuth provider. Also commonly called the "sub" or "Subject field" in OAuth protocols.
+    #   The type of this field is `String`.
+    # provider_type::
+    #   Denotes the OAuth identity provider that the user has authenticated with, e.g. Google, Facebook, GitHub etc.
+    #   The type of this field is `String`.
+    # session_token::
+    #   A secret token for a given Stytch Session.
+    #   The type of this field is `String`.
+    # session_jwt::
+    #   The JSON Web Token (JWT) for a given Stytch Session.
+    #   The type of this field is `String`.
+    # provider_values::
+    #   The `provider_values` object lists relevant identifiers, values, and scopes for a given OAuth provider. For example this object will include a provider's `access_token` that you can use to access the provider's API for a given user.
+    #
+    #   Note that these values will vary based on the OAuth provider in question, e.g. `id_token` is only returned by Microsoft.
+    #   The type of this field is `ProviderValues`.
+    # reset_sessions::
+    #   Indicates if all other of the User's Sessions need to be reset. You should check this field if you aren't using Stytch's Session product. If you are using Stytch's Session product, we revoke the User's other sessions for you.
```

Comment:
(docs problem) - we should really do a better job of explaining when this is true...